hera_mc/cm_hookup.py

```diff
@@ -4,52 +4,108 @@
 # Licensed under the 2-clause BSD license.
 
 """
-This is computes and displays part hookups.
+This finds and displays part hookups.
 
 """
 from __future__ import absolute_import, division, print_function
 from tabulate import tabulate
 import sys
 import numpy as np
-import os.path
+import os
+import time
+from astropy.time import Time, TimeDelta
 
-from hera_mc import mc, geo_location, correlator_levels, cm_utils, cm_handling
+from hera_mc import mc, geo_location, correlator_levels, cm_utils, cm_handling, cm_transfer
 from hera_mc import part_connect as PC
 import copy
 from sqlalchemy import func
 
 
+def get_parts_from_hookup(part_name, hookup_dict):
+    """
+    Retrieve the value for a part name from a hookup_dict
+    Parameters:
+    ------------
+    part_name:  string of valid part name in hookup_dict
+    hookup_dict:  hookup_dict to use
+
+    returns:
+        parts[hpn][pol] = (location, pam number)
+        location example: RI4A1E:A
+        pam number example: RCVR93:A (for a PAPER RCVR)
+        pam number example: PAM75101:B (for a HERA PAM)
+    """
+    if part_name not in hookup_dict['columns']:
+        return None
+    parts = {}
+    part_ind = hookup_dict['columns'].index(part_name)
+    for k, h in hookup_dict['hookup'].iteritems():  # iterates over parts
+        parts[k] = {}
+        for pol, p in h.iteritems():  # iterates over pols
+            pind = part_ind - 1
+            if pind < 0:
+                pind = 0
+            parts[k][pol] = (p[pind].upstream_part, p[pind].downstream_part)
+    return parts
+
+
 class Hookup:
     """
-    Class to find and display the signal path hookup.
+    Class to find and display the signal path hookup.  It only has three public methods:
```

Comment:
One of the follow-up commits should have updated the word "three" to "four". Going forward, probably best just to write "Its public methods are:".